ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/common/fragments/BaseListFragmentTest.java

```diff
@@ -0,0 +1,782 @@
+    }
+
+    @Test
+    public void addItemsToEndOfListTest() {
+        //Arrange
+        Whitebox.setInternalState(testBaseListFragment, "mAdapter", mockAdapter);
+        List<BaseListItem> items = new ArrayList<BaseListItem>();
+        doNothing().when(mockAdapter).addLoadMoreData(items);
+        doNothing().when(mockAdapter).hideLoadMoreProgress();
+        //Act
+        testBaseListFragment.addItemsToEndOfList(items);
+        //Assert
+        Assert.assertTrue(isMethodSuccessfullyRun);
+    }
+
+    @Test
+    public void addItemsToBeginningOfListTest() {
+        //Arrange
+        Whitebox.setInternalState(testBaseListFragment, "mAdapter", mockAdapter);
+        List<BaseListItem> items = new ArrayList<BaseListItem>();
+        doNothing().when(mockAdapter).addItems(items, 0);
+        doNothing().when(mockAdapter).hideLoadMoreProgress();
+        //Act
+        testBaseListFragment.addItemsToBeginningOfList(items);
+        //Assert
+        Assert.assertTrue(isMethodSuccessfullyRun);
+    }
+
+    @Test
+    public void addItemTest() {
+        //Arrange
+        Whitebox.setInternalState(testBaseListFragment, "mAdapter", mockAdapter);
+        BaseListItem item = Mockito.mock(BaseListItem.class);
+        doNothing().when(mockAdapter).addItem(item, 1);
+        //Act
+        testBaseListFragment.addItem(item, 1);
+        //Assert
+        Assert.assertTrue(isMethodSuccessfullyRun);
+    }
+
+    @Test
+    public void removeItemTest() {
+        //Arrange
+        Whitebox.setInternalState(testBaseListFragment, "mAdapter", mockAdapter);
+        doNothing().when(mockAdapter).removeItem(1);
+        //Act
+        testBaseListFragment.removeItem(1);
+        //Assert
+        Assert.assertTrue(isMethodSuccessfullyRun);
+    }
+
+    @Test
+    public void replaceItemTest() {
+        //Arrange
+        Whitebox.setInternalState(testBaseListFragment, "mAdapter", mockAdapter);
+        BaseListItem item = Mockito.mock(BaseListItem.class);
+        doNothing().when(mockAdapter).replaceItem(item, 1);
+        //Act
+        testBaseListFragment.replaceItem(item, 1);
+        //Assert
+        Assert.assertTrue(isMethodSuccessfullyRun);
+    }
+
+    @Test
+    public void scrollToEndOfListTest() throws Exception {
+        //Arrange
+        Whitebox.setInternalState(testBaseListFragment, "mAdapter", mockAdapter);
+        BaseListItem item = Mockito.mock(BaseListItem.class);
+        List<BaseListItem> items = new ArrayList<BaseListItem>();
+        items.add(item);
+        doReturn(items).when(mockAdapter).getData();
+        PowerMockito.whenNew(Handler.class).withNoArguments().thenReturn(handler);
+        //Act
+        testBaseListFragment.scrollToEndOfList();
+        //Assert
+        verify(handler).post(runnableCaptor.capture());
+        Runnable runnable = runnableCaptor.getValue();
+        runnable.run();
+        verify(mockMRecyclerView, times(1)).smoothScrollToPosition(anyInt());
```

Comment:
Please don't use generic matchers - any(), anyXXXX, etc
we know exactly what values we expect here
in case marcher is combined with verify we can use argument captor @Captor
Apply in all occurrences for all PRs

Reply:
done